Atomic/syscontainers.py

```diff
@@ -1388,7 +1388,7 @@ def _get_containers_at(self, checkouts, are_preinstalled, containers=None):
             fullpath = os.path.join(checkouts, x)
             if not os.path.exists(fullpath):
                 continue
-            if fullpath.endswith(".0") or fullpath.endswith(".1"):
+            if not os.path.islink(fullpath) and (fullpath.endswith(".0") or fullpath.endswith(".1")):
```

Comment:
Hmm, this might not be too relevant, but I am confused about the expression here. It would be great if you can clarify it a bit =).  I thought the actual folder for the container itself usually ends with .0 and .1 right? (and .0 or .1 is the deployment version).

Then If I understand this correctly, this check basically says, 1: if the path is not a symlink and 2: if the path is a container folder, we will skip the path. (but why do we want information only from a symlink path? shouldn't we care more about the actual existent path?)

The reason why I think folder name ends with .0 or .1 will be "actual container folder" (not symlink) is the following two lines: 
```
options["destination"] = options["destination"] or os.path.join(self._get_system_checkout_path(), "{}.{}".format(options["name"], options["deployment"]))
```
and 
```
        sym = os.path.join(self._get_system_checkout_path(), options["name"])
        if os.path.exists(sym):
            os.unlink(sym)
        os.symlink(options["destination"], sym)
```

which suggests the symlink is the one without the version number? Thanks in advance.

Reply:
we use only the name in the symlink as it doesn't have the additional ".0" and ".1" that are in the checkout directories.

I've modified the patch a bit and pushed a new version that includes also an inline comment.

Reply:
ah I think it starts to make sense now. Thanks for clarifying this. So to confirm I understand this, the reason why we only care about symlink is because it represents the current container, and it also eliminates the concern about the container with older versions (i.e the folder with both container versions in it. For instance, firewalld.0 and firewalld.1 can both appear in a same container directory, and we only want the current version of container, which is linked by the symlink) right?